src/extpipes/commands/deploy.py

```diff
@@ -1,187 +1,79 @@
-from datetime import datetime
-from typing import Dict, List, TypeVar
 
-from cognite.client.data_classes import ExtractionPipeline, ExtractionPipelineContact, ExtractionPipelineList
+from cognite.client.data_classes import ExtractionPipeline
 
 from .. import __version__
-from ..app_config import YesNoType
+from ..app_config import CommandMode
 from .base import CommandBase
 
-# type hints
-External_ids = List[str]
-
 
 class CommandDeploy(CommandBase):
-
-    # '''
-    #        .o8                       oooo
-    #       "888                       `888
-    #   .oooo888   .ooooo.  oo.ooooo.   888   .ooooo.  oooo    ooo
-    #  d88' `888  d88' `88b  888' `88b  888  d88' `88b  `88.  .8'
-    #  888   888  888ooo888  888   888  888  888   888   `88..8'
-    #  888   888  888    .o  888   888  888  888   888    `888'
-    #  `Y8bod88P" `Y8bod8P'  888bod8P' o888o `Y8bod8P'     .8'
-    #                        888                       .o..P'
-    #                       o888o                      `Y8P'
-    # '''
-    def command(self, automatic_delete: YesNoType) -> None:
-
-        # check click parameter and map from YesNoType to bool
-        # if provided they override configuration or defaults from yaml-config
-        if automatic_delete:
-            self.automatic_delete = automatic_delete == YesNoType.yes
-
-        # debug new features and override with cli-parameters
-        logging.debug(f"From cli: {automatic_delete=}")
-
+    def __init__(
+        self,
+        config_file,
+        command: CommandMode = CommandMode.DEPLOY,
+        debug: bool = False,
+        automatic_delete: bool = True,
+        dry_run: bool = False,
+    ):
+        super().__init__(config_file, command=command, debug=debug, dry_run=dry_run)
+        self.automatic_delete = automatic_delete
+
+    def command(self) -> None:
         # get existing extpipes
-        """
-        ### list and inspect existing extpipes
-        """
-        existing_extpipes = self.client.extraction_pipelines.list(limit=None)
-        # existing_extpipes[:2]
-
-        existing_extpipes_dict = {ep.external_id: ep for ep in existing_extpipes}
-
-        def get_new_ids(requested_ids: External_ids, existing_ids: External_ids) -> External_ids:
-            return list(set(requested_ids) - set(existing_ids))
-
-        def get_delete_ids(requested_ids: External_ids, existing_ids: External_ids) -> External_ids:
-            return list(set(existing_ids) - set(requested_ids))
-
-        # get existing -unused
-        """
-        existing_dict = {
-            ep.external_id: ep
-            for ep in self.client.extraction_pipelines.list(limit=-1)
-            # only manage dataops created extpipes and not manually created
-            # metadata can be None
-            if "dataops_created" in (ep.metadata if ep.metadata else {})
-        }
-        """
+        existing_extpipes = {ep.external_id: ep for ep in self.client.extraction_pipelines.list(limit=None)}
```

Comment:
Topic 5
I think this can be added back, but I would need to talk it out with you. I'm thinking it would be as easy as:
```suggestion
        existing_extpipes = {ep.external_id: ep for ep in self.client.extraction_pipelines.list(limit=None) if ep.metadata and "dataops_created" in ep.metadata}
```